_posts/2021-08-18-Day-8-Post.md

```diff
@@ -0,0 +1,123 @@
+```cpp
+#include <iostream>
+void convertFtoC(double & temperature)
+{
+    temperature= (temperature - 32)*(5.0/9.0);
+}
+
+int main(void)
+{
+    double temperature = 212.0;
+    std::cout<<"Original temperature is " <<temperature<<"F."<<std::endl;
+    convertFtoC(temperature);
+    std::cout<<"New temperature is " <<temperature<<"C."<<std::endl;
+    return 0;
+}
+```
+It is also possible to passing data by point. By passing the data with `*` in the signature, the address of the variable will be passed into the function. Passing by pointer is sometimes used to pass around arrays allocated with `new`, after which elements are accessed with `*`. However, this is awkward for other uses, where passing by references are preferred.
+
+## Const referencing
+
+A `const` reference can be passed into a function to make an imutable variable. Once a const variable is passed into a function, it cannot be changed within the function. However it is important to maintian *const correctness* when use const references. A `const` reference can accept both `const` and non-`const` data, but non-`const` reference can only take non-`const` objects. Therefore, a non-`const` reference cannot be made of a `const` object, you can't try to refer and change a `const` object. It's generally recommending to make referene arguments const if you do not have any intention of modifying the data, and is importing for maintaining the integrity of large data structures, like vectors and strings. It is much safer to write functions that do not mutate their input parameters, and to avoid global parameters.
+```cpp
+#include <iostream>
+double convertFtoC(double const temperature)
+{
+    return (temperature - 32)*(5.0/9.0);
+}
+
+int main(void)
+{
+    double temperatureF = 212.0;
+    double tempuratureC=convertFtoC(temperatureF);
+    std::cout<<"Original temperature is " <<temperatureF<<"F."<<std::endl;
+    std::cout<<"New temperature is " <<temperatureC<<"C."<<std::endl;
+    return 0;
+}
+```
+
+## Function Overloading
+
+In C++, we can have multiple functions of the same name, so long as the functions have different signatures. This is called function overloading Because the compiler will determine which function is used based on the data types and numbers of the argument. It is important to note that functions cannot be overloaded by return type, meaning you cannot have two functions that input the same data type, but return two different types of data.  For examples, we can have two `convertFtoC` functions, where one takes a `double` and the toher takes a `std::vector<double>`
```

Comment:
```suggestion
In C++, we can have multiple functions of the same name, so long as the functions have different signatures. This is called function overloading Because the compiler will determine which function is used based on the data types and numbers of the argument. It is important to note that functions cannot be overloaded by return type, meaning you cannot have two functions that input the same data type, but return two different types of data.  For examples, we can have two `convertFtoC` functions, where one takes a `double` and the other takes a `std::vector<double>`
```